solr/core/src/java/org/apache/solr/core/SchemaCodecFactory.java

```diff
@@ -124,18 +124,18 @@ public KnnVectorsFormat getKnnVectorsFormatForField(String field) {
             if (fieldType instanceof DenseVectorField) {
               DenseVectorField vectorType = (DenseVectorField) fieldType;
               String knnAlgorithm = vectorType.getKnnAlgorithm();
-              if (knnAlgorithm != null) {
-                if (knnAlgorithm.equals(DenseVectorField.HNSW_ALGORITHM)) {
-                  int maxConn = vectorType.getHnswMaxConn();
-                  int beamWidth = vectorType.getHnswBeamWidth();
-                  return new Lucene94HnswVectorsFormat(maxConn, beamWidth);
-                }
+              if (DenseVectorField.HNSW_ALGORITHM.equals(knnAlgorithm)) {
+                int maxConn = vectorType.getHnswMaxConn();
+                int beamWidth = vectorType.getHnswBeamWidth();
+                return new Lucene94HnswVectorsFormat(maxConn, beamWidth);
```

Comment:
With regards to including this in Solr: 9.1.1:  Doesn't `Lucene94HnswVectorsFormat` require Lucene 9.4?  9.1.1 will stay on Lucene 9.3 afaik.

Reply:
You are absolutely right!
Can't we then cherry-pick and change it to Lucene93HnswVectorsFormat for the 9.1.1 branch?
I think it's a nice to have bugfix!

Reply:
I would assume so! :)